fvm/src/kernel/default.rs

```diff
@@ -314,19 +314,19 @@ where
         Ok(k)
     }
 
-    fn block_read(&mut self, id: BlockId, offset: u32, buf: &mut [u8]) -> Result<i32> {
+    fn block_read(&mut self, id: BlockId, offset: u32, buf: &mut (impl AsMut<[u8]> + ?Sized)) -> Result<i32> {
```

Comment:
One would usually take `impl AsRef<[u8]>`:

- This makes it possible to pass a buffer by-value (if desired).
- `AsRef<U>` is implemented over `&T` where `T: AsRef<U> +?Sized`.
- This lets you drop the `?Sized`.

Reply:
I.e.

```rust
     fn block_read(&mut self, id: BlockId, offset: u32, buf: impl AsMut<[u8]>) -> Result<i32> { 
```